src/Analyzer.Utilities/JTokenExtensions.cs

```diff
@@ -13,107 +15,320 @@ namespace Microsoft.Azure.Templates.Analyzer.Utilities
+                var arrayMatch = jArrayRegex.Match(propertyToFind);
+                if (arrayMatch.Success)
+                {
+                    propertyToFind = arrayMatch.Groups["property"].Value;
+                }
+
                 bool foundChild = false;
                 Exception lastException = null;
-                var arrayMatch = jArrayRegex.Match(propertyName);
-
-                foreach (var child in token.Children())
+                foreach (var child in tokenContext.Token.Children())
                 {
                     if (child is JProperty childProperty)
                     {
-                        if (arrayMatch.Success && string.Equals(arrayMatch.Groups["property"].Value, childProperty.Name, StringComparison.OrdinalIgnoreCase))
+                        if (string.Equals(propertyToFind, childProperty.Name, StringComparison.OrdinalIgnoreCase))
                         {
-                            try
+                            // Found the target property.
+                            foundChild = true;
+                            if (arrayMatch.Success)
                             {
-                                var index = int.Parse(arrayMatch.Groups["index"].Value);
-                                token = childProperty.Value[index];
-                                foundChild = true;
+                                // Path is indexing into this array property.
+                                // Attempt to find the requested index.
+                                if (childProperty.Value is JArray childArray)
+                                {
+                                    var index = arrayMatch.Groups["index"].Value;
+                                    if (index == "*")
+                                    {
+                                        // Wildcard index - add all array elements and move to next path segment.
+                                        AddAllArrayElementsToQueue(jtokensToSearchFrom, childArray, tokenContext);
+                                    }
+                                    else
+                                    {
+                                        try
+                                        {
+                                            var intIndex = int.Parse(index);
+
+                                            if (intIndex < childArray.Count)
+                                            {
+                                                // If index is in-bounds, select that index.
+                                                EnqueueChildJToken(jtokensToSearchFrom, childArray[intIndex], tokenContext); 
+                                            }
+                                            else
+                                            {
+                                                // Otherwise, set the context to the array itself
+                                                // and set found to false so searching stops.
```

Comment:
I don't see how by setting found to false, it stops the searching (ie. where in this foreach loop does it make the search stop). Maybe I am not looking well. Also, could be better to put line 197 `foundChild = true;` in the if condition instead if you are just declaring the bool in line 188, then later putting true, then putting false. But again, I may be missing something.